src/coreclr/inc/daccess.h

```diff
@@ -2128,7 +2128,10 @@ inline void DACCOP_IGNORE(DacCopWarningCode code, const char * szReasonString)
 // Declare TADDR as a non-pointer type so that arithmetic
 // can be done on it directly, as with the DACCESS_COMPILE definition.
 // This also helps expose pointer usage that may need to be changed.
-typedef ULONG_PTR TADDR;
+#ifdef FEATURE_NATIVEAOT
```

Comment:
CoreCLR has a different (conflicting) `TADDR` definition in the same daccess.h file at the beginning.

Reply:
Why would it be a huge change?

Reply:
Seems like I got rid of the redefinition somewhere along the way in my WIP branch so I don't need this anymore. It will naturally go away with some bigger refactoring around GC headers that I still have to cleanup and submit...

The issue was that there are basically three compilation modes - CoreCLR, [Cross]DAC, and NativeAOT. NativeAOT traditionally preferred the standard C types, while that's not really an option for "target address" in the CoreCLR/[Cross]DAC model. Inadvertently defining it one way or the other was causing issues either with the CoreCLR build or the NativeAOT one.

Reply:
Damn... I hit it again on full rebuild. For posterity, this is the original error:
```
  D:\runtime\src\coreclr\inc\daccess.h(2131): error C2371: 'TADDR': redefinition; different basic types
  D:\runtime\src\coreclr\gc\env\gcenv.base.h(456): note: see declaration of 'TADDR'
```

Reply:
Ultimately it boils down to this:
- There is a type `__int3264` which is how `ULONG_PTR` is defined (at least for the non-DAC scenarios).
- There are two definitions of `TADDR` inside daccess.h that either have to match or they have to be wrapped in `#if !NATIVEAOT` (the first one is) and `#if NATIVEAOT` (the second one is not).
- If you define it as `ULONG_PTR` it will inevitably conflict with the definition in gcenv.base.h which uses `uintptr_t`.
- If you define it as `uintptr_t` it will conflict with the `!NATIVEAOT` definition at the beginning of daccess.h and likely break the cross-DAC.
- `gcenv.base.h` doesn't have `ULONG_PTR` so it would redefine it and match the Windows definition

Reply:
So in short words, `ULONG_PTR` and `uintptr_t` are not the same type on Windows?

Reply:
> So in short words, `ULONG_PTR` and `uintptr_t` are not the same type on Windows?

Correct.

Reply:
Would https://github.com/jkotas/runtime/commit/3530a59e59c71717d87e09589ff2b2378a62e716 fix the problem that you are seeing? I would rather get rid of the conflicting definitions once and for all.

Reply:
Seems to work for `clr+clr.aot` build 👍

Reply:
I have pushed the change into this PR.